File: Minimum Jumps To Reach The End/minJumpsToEnd.py

# Min Jumps Steps to reach end
def minJumpsToEnd(jumps):

    # Initialize Variable
    countSteps = 0
    maxDistance = 0
    currentDistance = 0

    for index in range(len(jumps)):

        # Since we need to minimize the steps, therefore
        # 1. First move to the current Distance index value
        # 2. From there, we will try to reach end
        if index > currentDistance:

            # Max Distance should be keep track
            currentDistance = maxDistance

            # Keeping track of Steps Count
            countSteps += 1

        # Keeping track what is the max distance we can reach from all the past current position
        maxDistance = max(maxDistance, index+jumps[index])

    return countSteps


if __name__ == "__main__":

    # input
    jumps = [2, 3, 1, 1, 4]

    print(minJumpsToEnd(jumps))

File: Minimum Jumps To Reach The End/minJumpsToEnd.cpp
#include <bits/stdc++.h>
using namespace std;
int minJumpsToEnd(int arr[], int n) {
    // Function to calculate Min jumps to reach end
    int jumps[n + 1];

    if (n == 0 || arr[0] == 0)  // Corner Case
        return INT_MAX;

    jumps[0] = 0;
    for (int i = 1; i < n; i++) {
        jumps[i] = INT_MAX;
        for (int j = 0; j < i; j++) {
            if (i <= j + arr[j] && jumps[j] != INT_MAX) {
                jumps[i] = min(jumps[i], jumps[j] + 1);
                break;
            }
        }
    }
    return jumps[n - 1];
}
int main() {
    int n = 5; // Size of Array
    int arr[] = {2, 3, 1, 1, 4};
    cout << minJumpsToEnd(arr, n)<<endl;
    return 0;
}